src/lib/serviceRoutinesV2/getSubscription.cpp

```diff
@@ -0,0 +1,66 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Orion dev team
+*/
+
+#include "serviceRoutinesV2/getSubscription.h"
+
+#include <string>
+#include <vector>
+
+#include "apiTypesV2/Subscription.h"
+#include "common/JsonHelper.h"
+#include "common/string.h"
+#include "mongoBackend/mongoGetSubscription.h"
+#include "ngsi/ParseData.h"
+#include "rest/ConnectionInfo.h"
+#include "rest/OrionError.h"
+
+/* ****************************************************************************
+*
+* getSubscription -
+*
+* GET /v2/subscription/<id>
+*
+*/
+std::string getSubscription
+(
+    ConnectionInfo*            ciP,
+    int                        components,
+    std::vector<std::string>&  compV,
+    ParseData*                 parseDataP
+)
+{
+
+  ngsiv2::Subscription sub;
+  std::string          idSub = compV[2];
+  OrionError           oe    = mongoGetSubscription(sub, idSub , ciP->uriParam, ciP->tenant);
```

Comment:
Very elegant like this.
But, as I mentioned, if somehow this could work by reference ...